src/tools/dart-compile.md

```diff
@@ -0,0 +1,257 @@
+```
+
+### Known limitations
+
+The `exe` and `aot-snapshot` subcommands have some known limitations:
+
+No cross-compilation support ([issue 28617][])
+: The compiler supports creating machine code only for
+  the operating system it’s running on.
+  You need to run the compiler three times —
+  on macOS, Windows, and Linux —
+  to create executables for all three operating systems.
+  A workaround is to use a CI (continuous integration) provider
+  that supports all three operating systems.
+
+No signing support ([issue 39106][])
+: The format of the executables isn’t compatible with
+  standard signing tools such as codesign and SignTool.
+
+No support for dart:mirrors and dart:developer
+: For a complete list of the core libraries you can use,
+  see the **All** and **AOT** entries in the
+  [table of core Dart libraries](/guides/libraries).
+
+[issue 28617]: https://github.com/dart-lang/sdk/issues/28617
+[issue 39106]: https://github.com/dart-lang/sdk/issues/39106
+
+{{site.alert.tip}}
+  If one of these issues is important to you,
+  let the Dart team know by adding a "thumbs up" to the issue.
+{{site.alert.end}}
+
+{% comment %}
+  TODO: Get info from https://github.com/dart-lang/sdk/wiki/Snapshots
+
+  kernel: 
+      Use to package up an app into a single file and
+      reduce startup time.
+
+  jit:
+      To run the program, use `dart run myapp.jit`.
+{% endcomment %}
+
+
+## JIT snapshots (jit-snapshot) {#jit-snapshot}
+
+JIT snapshots include all the parsed classes and compiled code that's
+generated during a training run of a program.
+
+```terminal
+$ dart compile jit-snapshot bin/myapp.dart
+Compiling bin/myapp.dart to jit-snapshot file bin/myapp.jit.
+Hello world!
+$ dart run bin/myapp.jit
+Hello world!
+$
+```
+
+When running from an application snapshot,
+the Dart VM doesn't need to parse or compile classes and functions that
+were already used during the training run,
+so the VM starts running user code sooner.
+
+These snapshots are architecture specific,
+unlike snapshots produced using the
+[`kernel` subcommand](#kernel).
+
+## JavaScript (js) {#js}
+
+The `js` subcommand compiles Dart code to deployable JavaScript.
+Another Dart-to-JavaScript compiler, [`dartdevc`][],
+is for development use only.
+
+You usually use the [`webdev` tool][webdev] instead of
+directly using a Dart-to-JavaScript compiler.
+The [`webdev build`][] command, by default,
+produces the same output as `dart compile js`.
+The [`webdev serve`][] command uses `dartdevc` by default, but you can switch
+to it to `dart compile js` by using the `--release` flag.
+
+The `js` subcommand replaces the `dart2js` command.
+For more information, see the [`dart2js` documentation](/tools/dart2js).
+
+[`dartdevc`]: /tools/dartdevc
+[webdev]: /tools/webdev
+[`webdev build`]: /tools/webdev#build
+[`webdev serve`]: /tools/webdev#serve
+
+
+## Portable snapshots (kernel) {#kernel}
+
+Use the `kernel` subcommand to package up an app into a
+single, portable file.
```

Comment:
I think we should elaborate on what 'portable' means here, for example:

"...into a single, portable file, which can be run on all operating systems and CPU architectures."